pkg/controller/replication_controller.go

```diff
@@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
 		return nil
 	}
+	if len(controllers) > 1 {
+		rcNames := []string{}
+		for _, controller := range controllers {
+			rcNames = append(rcNames, fmt.Sprintf("%s/%s", controller.Namespace, controller.Name))
+		}
+		util.HandleError(fmt.Errorf("Multiple controllers matched for pod %s/%s.  Choosing the first of: %s.", pod.Namespace, pod.Name, strings.Join(rcNames, ", ")))
```

Comment:
It looks like at the backoff interval, but that's effectively forever.  Would be better to disable the other and log only once (as an event under that rc, probably).

> On Jun 26, 2015, at 7:00 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/controller/replication_controller.go:
> 
> > @@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
> >  		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
> >  		return nil
> >  	}
> > +	if len(controllers) > 1 {
> > +		rcNames := []string{}
> > +		for _, controller := range controllers {
> > +			rcNames = append(rcNames, fmt.Sprintf("%s/%s", controller.Namespace, controller.Name))
> > +		}
> > +		util.HandleError(fmt.Errorf("Multiple controllers matched for pod %s/%s.  Choosing the first of: %s.", pod.Namespace, pod.Name, strings.Join(rcNames, ", ")))
> How often is this going to generate this error?
> 
> —
> Reply to this email directly or view it on GitHub.
>